core/src/main/java/org/apache/calcite/rel/rules/ProjectTableScanRule.java

```diff
@@ -103,12 +103,12 @@ protected static boolean test(TableScan scan) {
   protected void apply(RelOptRuleCall call, Project project, TableScan scan) {
     final RelOptTable table = scan.getTable();
     assert table.unwrap(ProjectableFilterableTable.class) != null;
-
-    final Mappings.TargetMapping mapping = project.getMapping();
-    if (mapping == null
-        || Mappings.isIdentity(mapping)) {
+    if (!project.isMapping()) {
       return;
     }
+    final Mappings.TargetMapping mapping = Project.getPartialMapping(
+            project.getInput().getRowType().getFieldCount(),
+            project.getProjects());
```

Comment:
Shall we consider update https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/interpreter/Bindables.java#L250 ?

If not, it might be only safe to pass an identity mapping to `BindableTableScan.create`. Because in my understanding, current `BindableTableScan` doesn't have functionality for filterable and projectable scan.

Reply:
I am wondering if you can open another jira ticket for this. Therefore you can open a pull request for this. Thanks~

Reply:
Just as the doc says:
> Planner rule that converts a {@link Project}
    on a {@link org.apache.calcite.rel.core.TableScan}
    of a {@link org.apache.calcite.schema.ProjectableFilterableTable}
    to a {@link org.apache.calcite.interpreter.Bindables.BindableTableScan}.

If the BindableTableScan does not support project expressions push down, we should fix it firstly.